compiler/rustc_mir_build/src/builder/scope.rs

```diff
@@ -822,6 +832,15 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                         });
                         block = next;
                     }
+                    DropKind::ForLint => {
```

Comment:
Previously we were inserting a drop terminator for tail calls. Not certain if this is ever reachable, but it's more correct.